pkg/planner/core/casetest/hint/hint_test.go

```diff
@@ -366,3 +366,11 @@ func TestHints(t *testing.T) {
 		tk.MustQuery("show warnings").Check(testkit.Rows(output[i].Warn...))
 	}
 }
+
+func TestHintInReplace(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("create table t(d date)")
+	tk.MustExec("replace /*+ SET_VAR(sql_mode='ALLOW_INVALID_DATES') */ into t values ('2004-04-31');")
+}
```

Comment:
Move to integration test?

Reply:
In the original PR https://github.com/pingcap/tidb/pull/50007, this test is actually in the integration test (see https://github.com/pingcap/tidb/pull/50007/files#diff-7b1a286bc1ed3a6acd2b0382122827f63b4fb207134dacd208c4f55898305c9dR1). 

For the cherry-pick in release-7.5, the mysql-tester doesn't have https://github.com/pingcap/mysql-tester/pull/109, so it'll report grammar error (because the mysql-tester is using an older version of parser).

In https://github.com/pingcap/mysql-tester/pull/109, the dependency of TiDB parser is removed, so that integration test on master branch works fine.